src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -19,11 +19,13 @@ Namespace Microsoft.CodeAnalysis.Semantics
                             Return OperationKind.CompoundAssignmentExpression
                         End If
                     Case BoundKind.UserDefinedBinaryOperator
-                        Dim rightOperatorBinary As IBinaryOperatorExpression = DirectCast(Create(value.Right), IBinaryOperatorExpression)
+                        Dim rightOperatorBinary As BoundUserDefinedBinaryOperator = DirectCast(value.Right, BoundUserDefinedBinaryOperator)
+
                         ' It is not permissible to access the Left property of a BoundUserDefinedBinaryOperator unconditionally,
                         ' because that property can throw an exception if the operator expression is semantically invalid.
-                        ' Fetching the left operand through IBinaryOperatorExpression is safe.
-                        If rightOperatorBinary.LeftOperand Is value.LeftOnTheRightOpt Then
+                        ' get it through helper method
+                        Dim leftOperand = GetUserDefinedBinaryOperatorChild(rightOperatorBinary, 0)
```

Comment:
>GetUserDefinedBinaryOperatorChild(rightOperatorBinary, 0) [](start = 42, length = 57)

It looks like this is still projecting to IOperation, we should stay in bound node universe.